src/mpid/ch4/shm/ipc/src/globals.c

```diff
@@ -28,22 +28,25 @@ MPIDI_IPCI_global_t MPIDI_IPCI_global;
 MPL_dbg_class MPIDI_IPCI_DBG_GENERAL;
 #endif
 
-bool MPIDI_IPCI_is_repeat_addr(void *addr)
+int MPIDI_IPCI_is_repeat_addr(void *addr)
 {
     if (!MPIR_CVAR_CH4_IPC_MAP_REPEAT_ADDR) {
-        return false;
+        return 0;
     }
-
-    static void *repeat_addr[10] = { 0 };
+#define MAX_REPEAT_ADDRS 10
```

Comment:
not specific to this change but we should change this value to a power of 2 for nicer code generation especially with the modulo operation